### PR TITLE
Added support for Cilium network policy

### PR DESCRIPTION
**Pull request by wazir-ahmed**
* Support for importing Cilium Network Policy via KArmor
* Modified `vm.PolicyHandling()` and `vm.sendPolicyOverHTTP()`

Signed-off-by: Wazir Ahmed <wazir@accuknox.com>

### CODE REVIEW

File: vm/policy.go [hunk 1]
```diff
@@ -15,10 +15,15 @@ import (
 
 	tp "github.com/kubearmor/KubeArmor/KubeArmor/types"
 	pb "github.com/kubearmor/KubeArmor/protobuf"
+	"github.com/kubearmor/kubearmor-client/cilium"
```
Comment (daemon1024) [inline vm/policy.go:18]: Can we directly import the typesfrom the cilium package?

https://github.com/cilium/cilium/blob/master/pkg/k8s/apis/cilium.io/v2/cnp_types.go#L32

File: go.sum [hunk 2]
```diff
@@ -137,7 +137,6 @@ github.com/cilium/ebpf v0.0.0-20200110133405-4032b1d8aae3/go.mod h1:MA5e5Lr8slmE
 github.com/cilium/ebpf v0.0.0-20200702112145-1c8d4c9ef775/go.mod h1:7cR51M8ViRLIdUjrmSXlK9pkrsDlLHbO8jiB8X8JnOc=
 github.com/cilium/ebpf v0.2.0/go.mod h1:To2CFviqOWL/M0gIMsvSMlqe7em/l1ALkX1PyjrX2Qs=
 github.com/cilium/ebpf v0.4.0/go.mod h1:4tRaxcgiL706VnOzHOdBlY8IEAIdxINsQBcU4xJJXRs=
-github.com/clarketm/json v1.17.1 h1:U1IxjqJkJ7bRK4L6dyphmoO840P6bdhPdbbLySourqI=
```
Comment (daemon1024) [inline go.sum:140]: Not sure if these `go.sum` changes are related to other changes in the PR. Can you please check once?

File: cilium/cidr.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package cilium
+
+import "net"
+
```
Comment (seswarrajan) [inline cilium/cidr.go:4]: // CIDR -- Brief comments

File: cilium/cidr.go [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+type CIDR struct {
+	*net.IPNet
+}
+
+type CIDRSlice []string
```
Comment (seswarrajan) [inline cilium/cidr.go:9]: // CIDRSlice -- Brief comments

**Review by seswarrajan — APPROVED** [review]

**Review by nyrahul — CHANGES_REQUESTED** [review]
Two comments:
1. Please reuse the types from cilium by directly importing it.
2. Fix the lint issues.

Thanks

File: go.sum (hunk 2 above)
Comment (wazir-ahmed) [reply go.sum:140]: This is not related to my changes. Even without my changes, if I do `go mod tidy`, this change happens. Looks like this due not including the changes in `go.sum` during previous commits.

File: cilium/cidr.go (hunk 3 above)
Comment (wazir-ahmed) [reply cilium/cidr.go]: Imported `github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2` directly. So this comment is not applicable now.

File: cilium/cilium.go [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package cilium
```
Comment (daemon1024) [inline cilium/cilium.go:1]:
```suggestion
package vm
```
let's declare this under vm package only, `vm/cilium.go`

Comment (wazir-ahmed) [reply cilium/cilium.go]: Done.

**Review by daemon1024 — APPROVED** [review]

**Review by nyrahul — APPROVED** [review]
LGTM

**Pull request merged**